### PR TITLE
ISPN-11263 CDI extension doc updates

### PR DESCRIPTION
**Pull request by oraNod**
https://issues.redhat.com/browse/ISPN-11263

### CODE REVIEW

File: documentation/src/main/asciidoc/topics/proc_cdi_injecting_remote.adoc [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+...
+import javax.enterprise.context.ApplicationScoped;
+
+public class Config {
+
+    @RemoteGreetingCache
+    @Produces
+    @ApplicationScoped
+    public RemoteCacheManager defaultRemoteCacheManager() {
+        return new RemoteCacheManager(localhost, 1544);
```
Comment (oraNod) [inline documentation/src/main/asciidoc/topics/proc_cdi_injecting_remote.adoc]: @danberindei should this be `RemoteCacheManager()` without the (string, int)?

as you say there is no configuration with remote caches. `RemoteCacheManager()` looks up the list of servers in a file hotrod-client.properties in the classpath

Comment (danberindei) [reply documentation/src/main/asciidoc/topics/proc_cdi_injecting_remote.adoc]: Yes, that's I wanted to say initially, but a more appropriate example would be
```
ConfigurationBuilder builder = new ConfigurationBuilder();
builder.addServer().host("localhost").port(11222);
return new RemoteCacheManager(builder.build());
```

File: documentation/src/main/asciidoc/topics/proc_cdi_injecting_embedded.adoc [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+@Target({ElementType.FIELD, ElementType.PARAMETER, ElementType.METHOD})
+@Retention(RetentionPolicy.RUNTIME)
+@Documented
+public @interface GreetingCache { <1>
+}
+----
++
+<1> creates a `@GreetingCache` qualifier.
++
+. Add a producer class that defines cache configuration objects.
```
Comment (danberindei) [inline documentation/src/main/asciidoc/topics/proc_cdi_injecting_embedded.adoc]:
```suggestion
. Add a producer method that defines the cache configuration.
```

File: documentation/src/main/asciidoc/topics/proc_cdi_injecting_embedded.adoc [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+                        .size(1000)
+                    .build();
+    }
+}
+----
++
+<1> names the cache to inject.
+<2> adds the cache qualifier.
++
+. Configure transport for clustered cache modes, if required.
```
Comment (danberindei) [inline documentation/src/main/asciidoc/topics/proc_cdi_injecting_embedded.adoc]:
```suggestion
. Add a producer method that creates a clustered cache manager, if required
```

File: documentation/src/main/asciidoc/topics/proc_cdi_injecting_embedded.adoc [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
++
+[source,java]
+----
+...
+package org.infinispan.configuration.global.GlobalConfigurationBuilder;
+
+public class Config {
+
+    @GreetingCache
```
Comment (danberindei) [inline documentation/src/main/asciidoc/topics/proc_cdi_injecting_embedded.adoc]: I'd add a note here too, saying smth like 

```
<1> adds the cache qualifier. Normally an application does not need more than one cache manager, and the cache manager producer should have no qualifier.
```

Or maybe it's better to mention this in the existing note 2?

**Comment by danberindei** [comment]
Merged, thanks Don!

**Pull request merged**